consult-yasnippet.el

```diff
@@ -0,0 +1,158 @@
+;;; consult-yasnippet.el --- Provides the command `consult-yasnippet' -*- lexical-binding: t; -*-
+
+;; Author: Mohsin Kaleem, Consult and Selectrum contributors
+;; Maintainer: Daniel Mendler
+;; Created: 2021
+;; License: GPL-3.0-or-later
+;; Version: 0.1
+;; Package-Requires: ((consult "0.1") (yasnippet "0.14") (emacs "26.1"))
+;; Homepage: https://github.com/minad/consult
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Provides the command `consult-yasnippet' which presents all the snippet
+;; expansions available in the current buffer with in-buffer previews.
+;;
+;; This is an extra package, to avoid loading yasnippet in the base consult
+;; package..
+
+
+;;; Code:
+
+(require 'consult)
+(require 'yasnippet)
+
+(defgroup consult-yasnippet nil
+  "Consulting `completing-read' for yasnippet snippets."
+  :group 'consult
+  :prefix "consult-yasnippet-")
+
+(defvar consult-yasnippet--snippets nil
+  "Snippet collection for current `consult-snippet' session.")
+
+(defvar consult-yasnippet--buffer nil
+  "The buffer in which `consult-yasnippet' was begun")
+
+(defvar consult-yasnippet--region nil
+  "The position (a cons of the start and end `point's) of where `consult-yasnippet' was begun")
+
+(defvar consult-yasnippet--region-contents ""
+  "The original contents of `consult-yasnippet--region'.")
+
+(defun consult-yasnippet--expand-template (template)
+  "Expand the yasnippet template TEMPLATE at point."
+  (deactivate-mark)
+  (goto-char (car consult-yasnippet--region))
+  (when (not (string-equal "" consult-yasnippet--region-contents))
+    (push-mark (point))
+    (push-mark (cdr consult-yasnippet--region) nil t))
+
+  (yas-expand-snippet (yas--template-content template)
+                      nil nil
+                      (yas--template-expand-env template)))
+
+(defun consult-yasnippet--preview (template _)
```

Comment:
This function should return a lambda, which captures the preview state.

Reply:
Done